### ISSUE TITLE
About results of training and evaluation

### ISSUE THREAD

**Issue by kimwongyuda** [studio-ousia/ease#7]
Hi, first of all, thank you for your nice work.

I just trained monolingual ease model by running "train_monolingual_ease.sh" (I didn't modify anything) and evaluated trained model through below command
"python evaluation.py --model_name_or_path result/my-ease-bert-base-uncased --pooler cls_before_pooler --task_set sts".

However, the results has gap compared to those in paper.
![image](https://github.com/studio-ousia/ease/assets/35097915/e368a648-3f83-4333-8c88-7fe1a25fef09)
![image](https://github.com/studio-ousia/ease/assets/35097915/bcaf9e44-8b80-4717-8b28-c59533c05b13)

Could you tell me what i might have missed, which affects that gap?

**Comment by Sosuke115** [comment]
Hi

The issue might be due to the pretrained Wikipedia2Vec vectors. 
I used float32 for my experiments but converted them to float16 for Google Drive's storage limit when publishing them.
You could try training the Wikipedia2Vec vectors yourself.
https://github.com/studio-ousia/ease/blob/37a811ebf39f2cecb57386068944e8146f8138af/download_all.sh#L14
https://wikipedia2vec.github.io/wikipedia2vec/

**Comment by kimwongyuda** [comment]
I got wikipedia data (enwiki-20190120-pages-articles.xml.bz2) from https://archive.org/download/enwiki-20190120.
Then, I'm going to train Wikipedia2Vec vectors by following  https://wikipedia2vec.github.io/wikipedia2vec/commands/.

However, there are some hyperparameters.
So, could you let me know hyperparameters or setting or codes which you used for training float32 vectors?

Thank you.

**Comment by kimwongyuda** [comment]
> I got wikipedia data (enwiki-20190120-pages-articles.xml.bz2) from https://archive.org/download/enwiki-20190120. Then, I'm going to train Wikipedia2Vec vectors by following https://wikipedia2vec.github.io/wikipedia2vec/commands/.
> 
> However, there are some hyperparameters. So, could you let me know hyperparameters or setting or codes which you used for training float32 vectors?
> 
> Thank you.

I found that you used default hyperparameters in paper. Thank you.

**Comment by Sosuke115** [comment]
Yes. 
Note that I set the vector dimension (--dim-size) to 768 to align with BERT's hidden size, as mentioned in the paper.

**Comment by kimwongyuda** [comment]
I also share below result when using batchsize=64 (per_device_train_batch_size=8 & gradient_accumulation_steps=8) and enwiki.fp16.768.vec.

![image](https://github.com/studio-ousia/ease/assets/35097915/1f8dfc55-f3d7-4864-a247-6fe786e29b0b)

I just modified only batch size and gained improvement of avg 2.77 compared to batch size 128 (default batch size value in train_monolingual_ease.sh).

In contrast with the mention in SimCSE paper "We find that SimCSE is not sensitive to batch sizes as long as tuning the learning rates accordingly", it looks like EASE is sensitive to batch size.
But, it is possible that the appropriate learning rate was not used when batch size 128 was applied.

**Comment by Sosuke115** [comment]
Thank you for sharing the information. 
I've forgotten the detailed results for each hyperparameter when using EASE, but it may indeed be true that EASE is more sensitive to these hyperparameters than SimCSE.
As you've demonstrated, it seems there might be more optimal hyperparameters when using the float16 version of Wikipedia2Vec.

**Comment by kimwongyuda** [comment]
I trained entity vector by using wikipedia data in https://archive.org/download/enwiki-20190120, but the results based on this trained entity vector was not good compared to that in the paper.

Could you have the way for releasing pretrained entity vector?

Thank you.

**Comment by Sosuke115** [comment]
I see.
The pretrained vectors I used during my experiments are on the server that I may not be able to access at the moment. 
I will check later to see if I can access them.

Could you please show me the performance of EASE using that vector?
Also, since the performance might depend on the slight randomness of Wikipedia2Vec, could you please try training EASE with different hyperparameters?
(e.g. learning rate ∈ {3e − 05, 5e − 05},  batch size ∈ {64, 128}  )

Furthermore, there's a possibility that Wikipedia2Vec might not be the cause.
(At the moment, I'm not sure what could be the cause...)

**Comment by kimwongyuda** [comment]
Oh, I understand your current situation.

I used lr = 3e-05 & batch size = 64 and the result is below.
![image](https://github.com/studio-ousia/ease/assets/35097915/bb4cb511-a5c0-465c-865d-21a6f3604dd2)

Also, when using lr = 3e-05 & batch size = 128, the result is below.
![image](https://github.com/studio-ousia/ease/assets/35097915/acd15b69-29ef-46ae-bfb0-0ecbd6e9e943)

You don't need to overdo it for accessing pertrained vectors !!
I really appreciate your opinions and feedbacks.
Thank you!

**Comment by Sosuke115** [comment]
I see.
Given such a significant performance difference with the fp32 vector, it's unlikely that the pretrained Wikipedia2Vec vector is the cause.
I'll inspect the execution settings later to determine what might be causing the issue.
Thank you for sharing this information.

**Comment by Sosuke115** [comment]
@kimwongyuda 

I have identified a potential cause.
Please try selecting `--pooler_type cls` during training, and `--pooler_type cls_before_pooler`during evaluation.
In monolingual EASE, a linear layer should be added during training only, the same as in Unsupervised SimCSE.

>  We add a linear layer after the output sentence embeddings only during training, as in Gao et al. (2021).

**Comment by kimwongyuda** [comment]
Thank you for your help!

I think we can close this issue.

Thank you!
